gen-val/src/crypto/src/NIST.CVP.ACVTS.Libraries.Crypto/XECDH/X25519.cs

```diff
@@ -0,0 +1,66 @@
+using System.Numerics;
+using NIST.CVP.ACVTS.Libraries.Crypto.Common.Asymmetric.XECDH;
+using NIST.CVP.ACVTS.Libraries.Crypto.Common.Asymmetric.XECDH.Enums;
+using NIST.CVP.ACVTS.Libraries.Math;
+using NIST.CVP.ACVTS.Libraries.Math.Entropy;
+
+namespace NIST.CVP.ACVTS.Libraries.Crypto.XECDH
+{
+    public class X25519 : XecdhBase
+    {
+        public X25519(Curve curveName, BigInteger p, BigInteger g, int bits, int a24, EntropyProviderTypes entropyType) :
```

Comment:
It would be better to provide the `IEntropyProvider` instead of the `EntropyProviderTypes` enum. This will enable the `TestableEntropyProvider` to be used if it is ever needed. In the `TestableEntropyProvider` you provide the outputs ahead of time to have deterministic outputs. That wouldn't be possible here because it couldn't be loaded from what calls this class.

Reply:
I can do that. Right now the XecdhBase class has an AddEntropy function that provides entropy through to the IEntropyProvider instance that's created here. This is pretty much how EdDSA does it. I changed it now, to remove the AddEntropy function and let the user provide IEntropyProvider directly  to IXecdhFactory. Let me know what you think: https://github.com/usnistgov/ACVP-Server/pull/391/commits/6974f0c88d51a55b2934fe5253ab71d0e8c9e8a5

Reply:
Resolved in https://github.com/usnistgov/ACVP-Server/pull/391/commits/38233798ca8db52dc7eccdcfc7619c2334146c0a